sips/committee_consensus.md

```diff
@@ -0,0 +1,480 @@
+type Committee interface {
+	// Initializes and starts the runner for the duties for the given slot
+  	StartDuty(duty CommitteeDuty) error
+	// ProcessMessage processes a message routed to this Committee
+	ProcessMessage(msg *types.SignedSSVMessage)
+}
+
+type Committee struct {
+	Runners                 map[spec.Slot]*CommitteeRunner
+	Operator                types.Operator
+	SignatureVerifier       types.SignatureVerifier
+	CreateRunnerFn          func() *CommitteeRunner
+	// Save the highest slot a validator attested to
+	HighestAttestingSlotMap map[types.ValidatorPK]spec.Slot
+}
+
+// Duty interface
+type Duty interface {
+	DutySlot() spec.Slot
+}
+
+// CommitteeDuty aggregates attesting and sync committee duties
+type CommitteeDuty struct {
+	Slot         spec.Slot
+	BeaconDuties []*BeaconDuty
+}
+
+// CommitteeRunner manages the duty cycle for a certain slot
+type CommitteeRunner interface {
+	// Start the duty lifecycle for the given slot. Emits a message.
+    StartDuty(duty Duty) error
+	// Processes cosensus message
+    ProcessConsensus(consensusMessage *qbft.Message) error
+	// Processes a post-consensus message
+    ProcessPostConsensus(msg PartialSignatureMessages) 
+}
+
+// Committee runner implements the old runner interface
+type CommitteeRunner struct {
+	// Important fields only
+	Shares          map[ValidatorPubkey]Share
+    Operator        Operator
+	QBFTController *qbft.Controller
+	BeaconNetwork  *types.BeaconNetwork
+}
+
+// BeaconVote is the consensus data
+type BeaconVote struct {
+    BlockRoot         phase0.Root
+    Source            phase0.Checkpoint
+    Target            phase0.Checkpoint
+}
+
+func ConstructAttestation(vote BeaconVote, duty AttesterDuty) Attestation {
+    bits := bitfield.New(duty.CommitteeLength)
+    bits.Set(duty.ValidatorCommitteeIndex)
+
+    return Attestation{
+        Slot: duty.Slot,
+        BlockRoot: vote.BlockRoot,
+        Source: vote.Source,
+        Target: vote.Target,
+        CommiteeIndex: duty.CommitteeIndex,
+        AggregationBits: bits,
+    }
+}
+
+func ConstructSyncCommittee(vote BeaconVote, duty AttesterDuty) SyncCommitteeMessage{
+    return SyncCommitteeMessage
+    {
+        Slot: duty.Slot,
+        BlockRoot: vote.BlockRoot,
+        ValidatorIndex: duty.ValidatorIndex,
+    }
+}
+```
+#### PartialSignatureMessages
+
+The current structure that we have in code can be kept with small changes.
+
+```go
+type PartialSignatureMessages struct {
+	Type     PartialSigMsgType
+	Slot     phase0.Slot
+	Messages []*PartialSignatureMessage
+}
+
+// PartialSignatureMessage is a msg for partial Beacon chain related signatures (like partial attestation, block, randao sigs)
+type PartialSignatureMessage struct {
+	PartialSignature Signature `ssz-size:"96"` // The Beacon chain partial Signature for a duty
+	SigningRoot      [32]byte  `ssz-size:"32"` // the root signed in PartialSignature
+	Signer         OperatorID
```

Comment:
this can be removed in a later stage, @GalRogozinski please add an issue for it

Reply:
part of https://github.com/ssvlabs/ssv-spec/issues/368

Reply:
there is a actually an issue...
you need both signer and validator index now since different validators in the committee may sign the same root